manager/scheduler/scheduler.go

```diff
@@ -74,6 +74,12 @@ func (s *Scheduler) setupTasksList(tx store.ReadTx) error {
 			continue
 		}
 
+		// Also ignore tasks that have not yet been assigned but desired state is beyond TaskStateRunning
+		// This can happen if you update, delete or scale down a service before its tasks were assigned.
+		if t.Status.State == api.TaskStatePending && t.DesiredState > api.TaskStateRunning {
```

Comment:
Why is this comparison not `t.Status.State <= api.TaskStatePending`

Reply:
Do you mean `t.Status.State <= api.TaskStateAssigned` ?

Reply:
The < is already handled in the condition above @nishanttotla

Reply:
@anshulpundir sorry maybe I'm missing something, but for `case api.EventUpdateTask:`, I don't see a specific condition about this.

Reply:
My bad, please ignore my comments. I got confused between two changes.